sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -467,25 +467,13 @@ class ResolveSessionCatalog(
         v1TableName.asTableIdentifier,
         partitionSpec)
 
-    case ShowColumnsStatement(tbl, ns) =>
-      if (ns.isDefined && ns.get.length > 1) {
-        throw new AnalysisException(
-          s"Namespace name should have only one part if specified: ${ns.get.quoted}")
-      }
```

Comment:
Note that this is not hit anymore now that namespace is always added to the table name in `AstBuilder` so that v2 tables can be resolved.

If there are multi parts in the namespace name, it will fail while looking up in the session catalog: `The namespace in session catalog must have exactly one name part`. I added a test for this below.